### PR TITLE
Add API documentation

### LINKED ISSUE #327: Document API Endpoints
#### Proposal

Since we are implementing an API, we should document what the endpoints are

### PR DESCRIPTION
**Pull request by dtomcej**
This PR:

- Adds API documentation for the newly added Controller API

The documentation can be viewed via the netlify preview 😄 

Fixes #327

### CODE REVIEW

File: docs/content/api.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## `/api/status/node/{node}/configuration`
+
+This endpoint provides raw json of the current configuration on the Maesh node with the pod name given in `{node}`.
+This endpoint provides a 404 response if the pod cannot be found, or other non-200 status codes on other errors.
+If errors are encountered, the error will be returned in the body, and logged on the controller.
+
+## `/api/status/readiness`
+
+This endpoint returns a 200 response if the controller successfully deployed a configuration to a Maesh node, and Maesh is ready for use.
```
Comment (jlevesy) [inline docs/content/api.md]: WDYT ?

```suggestion
This endpoint returns a 200 response if the controller successfully deployed a configuration to all Maesh nodes, and Maesh is ready for use.
```

File: docs/content/api.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+Otherwise, it will return a 500.
+
+## `/api/log/deploylog`
```
Comment (jlevesy) [inline docs/content/api.md]: `/api/log/deploylog` feels a bit redundant to me.
I'm asking a feature here haha, but WDYT of `/api/log` or `/api/log/deployment` instead ?

File: docs/content/api.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+!!! Note
+    This may change each request, as it is a live data structure.
+
+## `/api/status/nodes`
+
+This endpoint provides a json array containing some details about the readiness of the Maesh nodes visible by the controller
+This endpoint will still return a 200 if there are no visible nodes.
+
+## `/api/status/node/{node}/configuration`
```
Comment (jlevesy) [inline docs/content/api.md]: Not convinced by this change, but I would be nice if we had a hint about using the pod name directly.

```suggestion
## `/api/status/node/{maesh-pod-name}/configuration`
```

File: docs/content/api.md (hunk 2 above)
Comment (SantoDE) [reply docs/content/api.md]: Agree on that

**Pull request merged**